packages/core/src/external/aws/lambda-logic/bundle-to-html-derm.ts

```diff
@@ -0,0 +1,2688 @@
+      observation => dayjs(observation.effectiveDate).format(ISO_DATE) === date
+    );
+
+    return {
+      effectiveDate: diastolic.effectiveDate,
+      vitalsValue: `${systolicValue?.vitalsValue} - ${diastolic.vitalsValue}`,
+    };
+  });
+
+  const observationsDiastolicAscending = cloneDeep(observationDiastolicObjects).sort((a, b) => {
+    return dayjs(a.effectiveDate).isBefore(dayjs(b.effectiveDate)) ? -1 : 1;
+  });
+
+  const observationsSystolicAscending = cloneDeep(observationSystolicObjects).sort((a, b) => {
+    return dayjs(a.effectiveDate).isBefore(dayjs(b.effectiveDate)) ? -1 : 1;
+  });
+
+  const chartDiastolicData = {
+    labels: observationsDiastolicAscending.map(obs => obs.effectiveDate),
+    data: observationsDiastolicAscending.map(obs => parseFloat(obs.vitalsValue)),
+    min: Math.floor(
+      Math.min(...observationsDiastolicAscending.map(obs => parseFloat(obs.vitalsValue))) - 1
+    ),
+    max: Math.ceil(
+      Math.max(...observationsDiastolicAscending.map(obs => parseFloat(obs.vitalsValue))) + 1
+    ),
+  };
+
+  const chartSystolicData = {
+    labels: observationsSystolicAscending.map(obs => obs.effectiveDate),
+    data: observationsSystolicAscending.map(obs => parseFloat(obs.vitalsValue)),
+    min: Math.floor(
+      Math.min(...observationsSystolicAscending.map(obs => parseFloat(obs.vitalsValue))) - 1
+    ),
+    max: Math.ceil(
+      Math.max(...observationsSystolicAscending.map(obs => parseFloat(obs.vitalsValue))) + 1
+    ),
+  };
+
+  const observationRows = combinedObjects
+    .map(obs => {
+      return `
+      <tr>
+      <td>${obs.effectiveDate}</td>
+      <td>${obs.vitalsValue}</td>
+      </tr>
+      `;
+    })
+    .join(" ");
+
+  const observationTableContents = `
+    <table>
+    <thead>
+    <tr>
+    <th style="width: 50%">Date</th>
+    <th style="width: 50%">Value</th>
+    </tr>
+    </thead>
+    <tbody>
+    ${observationRows}
+    </tbody>
+    </table>
+    `;
+
+  return { tableContent: observationTableContents, chartDiastolicData, chartSystolicData };
+}
+
+function createFromObservationVitalsSection(
+  observations: Observation[],
+  vitalTitle: string,
+  loincCode: string
+): {
+  section: string;
+  chartData: ChartData;
+} {
+  if (!observations) {
+    return { section: "", chartData: { labels: [], data: [] } };
+  }
+
+  const filteredObservations = observations.filter(observation => {
+    const observationDisplay = observation.code?.coding?.find(coding => {
+      const code = fetchCodingCodeOrDisplayOrSystem(coding, "code");
+      return code === loincCode;
+    });
+
+    return !!observationDisplay;
+  });
+
+  const observationsLast2Years = filteredObservations.filter(observation => {
+    return dayjs(observation.effectiveDateTime).isAfter(dayjs().subtract(2, "year"));
+  });
+
+  const observationsSortedByDate = observationsLast2Years.sort((a, b) => {
+    return dayjs(a.effectiveDateTime).isBefore(dayjs(b.effectiveDateTime)) ? 1 : -1;
```

Comment:
Prob a good idea to create a `sortDate(date1: ConfigType, date2: ConfigType, sortingOrder: 'asc" | "desc" = "asc")` function on `packages/shared`... this is duplicated multiple times on this file alone.

Could place on `src/common/date.ts`.